nmpolicy/internal/lexer/lexer_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 The NMPolicy Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lexer_test
+
+import (
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/lexer"
+)
+
+func TestLexer(t *testing.T) {
+	type expected struct {
+		tokens []lexer.Token
+		err    string
+	}
+
+	var tests = []struct {
+		expression string
+		expected   expected
+	}{
+		{"    ", expected{tokens: []lexer.Token{
```

Comment:
Please add tests with `+` and `|` as part of the expression. e.g "1|5", "12 5+7"

Reply:
This is the commit for numbers.

Reply:
I know, I want the tests to tests "1|5" and "12 5+7" fails tokenization. If you want to add it to a further commit it's fine.